assemblyline_core/dispatching/client.py

```diff
@@ -189,18 +189,21 @@ def service_finished(self, sid: str, result_key: str, result: Result,
         # most distant expiry time to prevent pulling it out from under another submission too early
         if result.is_empty():
             # Empty Result will not be archived therefore result.archive_ts drives their deletion
-            self.ds.emptyresult.save(
-                result_key, {"expiry_ts": result.archive_ts},
-                force_archive_access=self.config.datastore.ilm.update_archive)
+            self.ds.emptyresult.save(result_key, {"expiry_ts": result.archive_ts})
         else:
-            with Lock(f"lock-{result_key}", 5, self.redis):
-                old = self.ds.result.get(result_key, force_archive_access=self.config.datastore.ilm.update_archive)
+            while True:
+                old, version = self.ds.result.get(
```

Comment:
Maybe there is something that I'm not understanding, but if there is a version conflict, then this will become an infinite loop?

Reply:
No because the get operation will give you a new version to save as each time.